src/Dompdf.php

```diff
@@ -284,36 +284,40 @@ public function __construct($options = null)
      */
     private function setPhpConfig()
     {
-        if (sprintf('%.1f', 1.0) !== '1.0') {
-            $this->systemLocale = setlocale(LC_NUMERIC, "0");
-            setlocale(LC_NUMERIC, "C");
-        }
+        if (function_exists('ini_get') && function_exists('ini_set')) {
+            if (sprintf('%.1f', 1.0) !== '1.0') {
+                $this->systemLocale = setlocale(LC_NUMERIC, "0");
+                setlocale(LC_NUMERIC, "C");
+            }
 
-        $this->pcreJit = @ini_get('pcre.jit');
-        @ini_set('pcre.jit', '0');
+            $this->pcreJit = @ini_get('pcre.jit');
+            @ini_set('pcre.jit', '0');
 
-        $this->mbstringEncoding = mb_internal_encoding();
-        mb_internal_encoding('UTF-8');
+            $this->mbstringEncoding = mb_internal_encoding();
+            mb_internal_encoding('UTF-8');
+        }
```

Comment:
Why did you wrap this whole block in the condition instead of just the lines using `ini_get`/`ini_set`?

Reply:
because I assumed that was the best way to do this with the fewest number of lines added. I can add a few more conditionals so that only the 1-line `ini_set` and `ini_get` calls are wrapped, if others think that's better?

Reply:
The preference would be to change only the relevant lines. You're also disabling changes to locale and internal encoding, both of which can completely break rendering if configured in a way that's incompatible with Dompdf.